lib/rust/parser/src/metadata.rs

```diff
@@ -0,0 +1,74 @@
+//! Data associated with a syntax tree.
+
+use std::collections::BTreeMap;
+use std::str::FromStr;
+use uuid::Uuid;
+
+
+
+// ================
+// === Metadata ===
+// ================
+
+/// Attaches stable IDs to AST nodes, and associates properties with them.
+#[derive(Debug)]
+pub struct Metadata {
+    id_map: BTreeMap<(usize, usize), Uuid>,
```

Comment:
This type is confusing, its not obviosu what (usize, usize) is. Please refactor that to a well named data type instead.